### PR TITLE
Configurable paths

### PR DESCRIPTION
**Pull request by okcoker**
In my case, I have redirect rules in production that don't match up with the paths that are used in development via webpack. Since this plugin assumes and uses paths provided by the webpack configuration, I needed a way to change what the plugin outputs. 

This adds a couple options: 
- `outputPath`, so you can choose where to output the favicon files relative to webpack's output directory config value
- `publicPath`, so you can override the webpack compilation's `publicPath` and have different directory paths within the actual html, json that is spit out

### CODE REVIEW

File: src/loader.js [hunk 1]
```diff
@@ -18,11 +18,11 @@ module.exports = function (content) {
     context: getContext(this),
     content: msgpack.encode([content, query.options, pkg.version]), // hash must depend on logo + config + version
   }));
-
+  const outputPath = query.outputPath ? trailingSlash(query.outputPath) : null;
```
Comment (brunocodutra) [inline src/loader.js]: I believe the intention is to have `outputPath` fallback to `prefix` instead of `null`.

File: README.md [hunk 2]
```diff
@@ -113,6 +113,11 @@ plugins: [
     // Enable caching and optionally specify the path to store cached data
     // Note: disabling caching may increase build times considerably
     cache: true,
+    // Override the publicPath option usually read from webpack configuration
+    publicPath: '/static',
+    // The directory to output the assets relative to the webpack output dir.
+    // Relative string paths are allowed here ie '../public/static'.
```
Comment (brunocodutra) [inline README.md]: could you explain that `outputPath` defaults to `prefix` if not set?

File: test/outputpathoption.test.js [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+
+  await generate({
+    context: t.context.root,
+    output: {
+      path: dist,
+      publicPath: '/public/path',
+    },
+    plugins: [
+      new HtmlWebpackPlugin(),
+      new WebappWebpackPlugin({ logo, outputPath: customOutputPath }),
```
Comment (brunocodutra) [inline test/outputpathoption.test.js]: Minor: any reason why not inlining `/test/path` here?

File: test/publicpathoption.test.js [hunk 4]
```diff
@@ -0,0 +1,30 @@
...
+  console.log(dist);
+  await generate({
+    context: t.context.root,
+    output: {
+      path: dist,
+      publicPath: '/public/path',
+    },
+    plugins: [
+      new HtmlWebpackPlugin(),
+      new WebappWebpackPlugin({ logo, publicPath: '/public/path/deeper' }),
```
Comment (brunocodutra) [inline test/publicpathoption.test.js]: Would you mind changing this to a something unrelated to the actual `publicPath`?

**Review by brunocodutra — CHANGES_REQUESTED** [review]
Really appreciate your work!
I have just a couple minor comments, once you're able to address them I'll be happy to merge.

File: test/outputpathoption.test.js (hunk 3 above)
Comment (okcoker) [reply test/outputpathoption.test.js]: Ha good catch I was actually trying to write tests without really looking into how all the utils worked so I had it used in multiple places at one point. Will update this

File: test/publicpathoption.test.js (hunk 4 above)
Comment (okcoker) [reply test/publicpathoption.test.js]: Do you mean making it something like `/another/path/excluding/public`?

Comment (brunocodutra) [reply test/publicpathoption.test.js]: Yeah or `a/b/`, I mean anything that doesn't suggest it has to relate to the webpack `publicPath`

**Comment by okcoker** [comment]
@brunocodutra ready when you are!

**Review by brunocodutra — APPROVED** [review]

**Comment by brunocodutra** [comment]
Thanks again @okcoker

**Comment by okcoker** [comment]
@brunocodutra any plan on getting this released?

**Comment by brunocodutra** [comment]
Right now.

**Pull request merged**